sway/desktop/xwayland.c

```diff
@@ -121,9 +121,35 @@ static void unmanaged_handle_destroy(struct wl_listener *listener, void *data) {
 	wl_list_remove(&surface->map.link);
 	wl_list_remove(&surface->unmap.link);
 	wl_list_remove(&surface->destroy.link);
+	wl_list_remove(&surface->override_redirect.link);
 	free(surface);
 }
 
+static void handle_map(struct wl_listener *listener, void *data);
+
+struct sway_xwayland_view *create_xwayland_view(struct wlr_xwayland_surface *xsurface);
+
+static void unmanaged_handle_override_redirect(struct wl_listener *listener, void *data) {
+	struct sway_xwayland_unmanaged *surface =
+		wl_container_of(listener, surface, override_redirect);
+	struct wlr_xwayland_surface *xsurface = surface->wlr_xwayland_surface;
+
+	bool mapped = xsurface->mapped;
+	if (mapped) {
+		unmanaged_handle_unmap(&surface->unmap, NULL);
+	}
+
+	wl_list_remove(&surface->request_configure.link);
```

Comment:
One last comment: why do we need to handle this listener as a special case?

Reply:
I didn't find it in unmanaged_handle_destroy or unmanaged_handle_unmap and wasn't sure whether to add it in one of those. For this case it needs to be removed though, otherwise sway crashes when trying to move/close or otherwise interact with a view that switched from unmanaged to managed.
I can move if it to one of those functions if you think that's more appropriate.

Reply:
Ah, it's probably missing then and not crashing "by chance". If the listener is added in `create_xwayland_view`, it should be removed in `handle_destroy`. If it's added in `handle_map`, it should be removed in `handle_unmap`.

Reply:
Ok, I moved it to `unmanaged_handle_destroy` then. For managed views the listener was already handled correctly.